pallets/pp/src/lib.rs

```diff
@@ -99,19 +182,38 @@ where
         caller: &T::AccountId,
         gas_limit: Weight,
         code: &Vec<u8>,
+        _utxo_hash: H256,
+        _utxo_value: u128,
         data: &Vec<u8>,
     ) -> Result<(), &'static str> {
         let code = pallet_contracts_primitives::Code::Upload(Bytes(code.to_vec()));
         let endowment = pallet_contracts::Pallet::<T>::subsistence_threshold();
 
-        let _ = pallet_contracts::Pallet::<T>::bare_instantiate(
+        let res = match pallet_contracts::Pallet::<T>::bare_instantiate(
             caller.clone(),
             endowment * 100_u32.into(), // TODO
             gas_limit,
             code,
             data.to_vec(),
             Vec::new(),
             true, // enable debugging
+        )
+        .result
+        {
+            Ok(res) => res,
+            Err(e) => {
+                log::error!("Failed to instantiate contract: {:?}", e);
+                return Err("Failed to instantiate contract");
+            }
+        };
```

Comment:
I find the formatting here a bit confusing. Maybe using something like `base_instantiate(...).result.map_err({ log(); "blah" })?` would be a bit more concise?